clar2wasm/tests/wasm-generation/main.rs

```diff
@@ -0,0 +1,253 @@
+pub mod values;
+
+use clarity::vm::types::{
+    ASCIIData, BuffData, ListData, ListTypeData, OptionalData, ResponseData, SequenceData,
+    SequenceSubtype, StringSubtype, TupleData, TupleTypeSignature, TypeSignature, Value,
+};
+use proptest::prelude::*;
+
+pub fn prop_signature() -> impl Strategy<Value = TypeSignature> {
+    let leaf = prop_oneof![
+        Just(TypeSignature::IntType),
+        Just(TypeSignature::UIntType),
+        Just(TypeSignature::BoolType),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::BufferType(
+            s.try_into().unwrap()
+        ))),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::StringType(
+            StringSubtype::ASCII(s.try_into().unwrap())
+        ))),
+        // TODO: principal,
+        // TODO: string-utf8
+        // TODO: CallableType
```

Comment:
`CallableType` is only an intermediate type, and can never be a parameter type. When a function takes a trait for example, the initial caller must pass a principal literal to it. Since that's the case, I think you can just generate principals and not worry about `CallableType`.